checker/src/expression.rs

```diff
@@ -127,6 +127,18 @@ pub enum Expression {
         right: Box<AbstractDomain>,
     },
 
+    /// Either left or right without a condition to tell them apart.
+    /// This can happen when there is loss of precision because of a loop fixed point computation.
+    /// For instance inside a loop body after the second fixed point iteration, a counter may have
+    /// either its initial value or the value computed in the first loop body iteration and we
+    /// don't have a way to tell which value it is.
+    Joined {
```

Comment:
I can't figure out if I like this name or not. It's a little bit confusing, because normally if two domains were joined you would just have..the joined domain. But I can't figure out a name that makes more sense to me. But either way, the description here explains it well, so meh.

Reply:
If you look closely, you'll see that Expression is not a domain. Not just because it says so in the comments, but because, unlike ConstantDomain and IntervalDomain there is no Domain suffix.

The role of Expression is to keep track of that operations that resulted in the abstract value that refers to it. It therefore does make sense to me to name that Expression instance that results from a join operation after the operation.

I have to concede, however, that "Join" is probably more in keeping with the other enumeration labels than "Joined". So you actually do have a point here. I'll change it.